### PR TITLE
Add test case for getting version-id when upload objects  bucket

### PR DESCRIPTION
**Pull request by cfanz**
This test case is used for testing whether a version-id header is returned when upload files.

Related with [ceph #18935](https://github.com/ceph/ceph/pull/18935)

Signed-off-by: Xinying Song <songxinying@cloudin.cn>

### CODE REVIEW

File: s3tests/functional/test_s3.py [hunk 1]
```diff
@@ -9315,3 +9316,67 @@ def test_delete_tags_obj_public():
...
+    key_name = 'baz'
+    key = bucket.new_key(key_name)
+    key.set_contents_from_string(key_name)
+    assert_is_none(key.version_id)
+
+    # for versioning-suspended-bucket, no version-id should return.
+    bucket = get_new_bucket()
+    key_name = 'foo'
+    key = bucket.new_key(key_name)
+    key.set_contents_from_string(key_name)
```
Comment (cbodley) [inline s3tests/functional/test_s3.py]: it looks like this last part is missing the call to suspend versioning and the check for `key.version_id`

File: s3tests/functional/test_s3.py [hunk 2]
```diff
@@ -9315,3 +9316,67 @@ def test_delete_tags_obj_public():
...
+    bucket = get_new_bucket()
+    check_configure_versioning_retry(bucket, True, "Enabled")
+    key_name = 'bar'
+    key = bucket.new_key(key_name)
+    key.set_contents_from_string(key_name)
+    # list_verions will return an non-empty version-id
+    li = bucket.list_versions()
+    for k in li:
+        eq(key.version_id, k.version_id)
+    return
```
Comment (cbodley) [inline s3tests/functional/test_s3.py]: need to remove the return here

Comment (cfanz) [reply s3tests/functional/test_s3.py]: @cbodley  Oops! Sorry for my carelessness. Have fixed it.

**Comment by cbodley** [comment]
cherry-picked to ceph-master

**Pull request merged**